File: src/main/java/com/github/quiltservertools/wires/command/StaffChatCommand.java
package com.github.quiltservertools.wires.command;

import com.github.quiltservertools.wires.Wires;
import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.arguments.StringArgumentType;
import me.lucko.fabric.api.permissions.v0.Permissions;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.text.LiteralText;

import static net.minecraft.server.command.CommandManager.argument;
import static net.minecraft.server.command.CommandManager.literal;

public class StaffChatCommand {
    public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(literal("staffchat")
                .requires(scs -> Permissions.check(scs, "wires.staffchat", 2))
                .executes(ctx -> {
                    Wires.config.getStaffChat().toggle(ctx.getSource().getPlayer().getUuid());
                    ctx.getSource().sendFeedback(new LiteralText(Wires.config.getStaffChat().isInStaffChat(ctx.getSource().getPlayer().getUuid()) ? "Moved to staff chat" : "Moved to game chat"), false);
                    return 1;
                })
                .then(argument("message", StringArgumentType.greedyString()).executes(
                        ctx -> {
                            Wires.config.getStaffChat().sendMessage(ctx.getSource().getPlayer(), StringArgumentType.getString(ctx, "message"));
                            return 1;
                        }
                )));

    }
}

File: src/main/kotlin/net/quiltservertools/wires/command/StaffchatCommand.kt
package net.quiltservertools.wires.command

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.arguments.StringArgumentType
import com.mojang.brigadier.context.CommandContext
import net.minecraft.server.command.CommandManager
import net.minecraft.server.command.CommandManager.literal
import net.minecraft.server.command.ServerCommandSource
import net.minecraft.text.Text
import net.quiltservertools.wires.StaffChat
import net.quiltservertools.wires.util.Permissions.hasPermission

object StaffChatCommand {

    private const val name = "staffchat"

    fun register(dispatcher: CommandDispatcher<ServerCommandSource>) {
        dispatcher.register(literal(name)
            .requires {
                it.hasPermission(name)
            }
            .executes { ctx: CommandContext<ServerCommandSource> ->
                StaffChat.toggle(ctx.source.player!!.uuid)
                ctx.source.sendFeedback(
                    Text.literal(
                        if (StaffChat.isInStaffChat(ctx.source.player!!.uuid)
                        ) "Moved to staff chat" else "Moved to game chat"
                    ), false
                )
                1
            }
            .then(
                CommandManager.argument("message", StringArgumentType.greedyString())
                    .executes {
                        StaffChat.sendMessage(it.source.player!!, StringArgumentType.getString(it, "message"))
                        1
                    })
        )
    }
}